File: 프로그래머스/3/43162. 네트워크/네트워크.py
def solution(n, computers):
    answer = 0
    visited = [False for _ in range(n)]

    def dfs(index):
        nonlocal visited
        visited[index] = True     
        
        for i in range(n):
            if computers[index][i] == 1 and not visited[i]:
                visited[i] = True
                dfs(i)
    
    for i in range(n):
        if not visited[i]:
            answer += 1
            dfs(i)
        
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const visited = Array(n).fill(false);
    let answer = 0;
    
    for (let i = 0; i < n; i++) {
        if (!visited[i]) {
            visited[i] = true;
            answer++;
            dfs(i);
        }
    }
        
        
    function dfs(node) {
        const computer = computers[node];
        
        for (let i = 0; i < n; i++) {
            if (!visited[i] && computer[i] == 1) {
                visited[i] = true;
                dfs(i);
            }
        }
        
    }
    
    return answer;
}